libcodechecker/libhandlers/analyze.py

```diff
@@ -177,6 +184,58 @@ def add_arguments_to_parser(parser):
                                help="File containing argument which will be "
                                     "forwarded verbatim for Clang-Tidy.")
 
+    if is_ctu_capable():
+        ctu_opts = parser.add_argument_group('cross translation unit analysis')
+        ctu_mutex_group = ctu_opts.add_mutually_exclusive_group()
+        ctu_mutex_group.add_argument('--ctu',
+                                     action='store_const',
+                                     const=[True, True],
+                                     dest='ctu_phases',
+                                     default=argparse.SUPPRESS,
+                                     help="Perform cross translation unit "
+                                          "(CTU) analysis (both collect and "
+                                          "analyze phases) using default "
+                                          "<ctu-dir> for temporary output. "
+                                          "At the end of the analysis, the "
+                                          "temporary directory is removed.")
+        ctu_opts.add_argument('--ctu-dir',
+                              metavar='<ctu-dir>',
+                              default=os.path.join(
+                                  util.get_default_workspace(),
+                                  'ctu-dir'),
+                              help="Optional parameter defines the temporary "
+                                   "directory used between CTU phases.")
+        ctu_mutex_group.add_argument('--ctu-collect-only',
```

Comment:
Yes, it also raised my concerns after seeing that this concept doesn't really fit the current architecture. My problem was that while clangsa and clangtidy can be nicely separated along a multithreaded interface as these plugins can run separately, we cannot do this with CTU 1st and 2nd phase as the order is important (You are right in that clang-func-mapping is only for the 1st phase). Currently no 2-phased process is supported without creating a separate subcommand for it.

So about the subcommands: The original reason for putting these phases together was user convenience. It was an expectation from multiple sources who want to use this.
But: although putting 1st phase into log seems to be a solution, it is also raises confusion. Running the log is only necessary when you change your project structure like adding new sources. CTU collect (1st) phase on the other hand has to be run on any code change as function names and the ast can change anytime during development. This drastic workflow change would confuse the user I think. Based on this, a new, separate subcommand would be the cleanest, but again, it would be clangsa only, would rely on other subcommand's parametrization (for example if you use reparse here, you should use reparse there...). To be honest, none of the proposed solutions seem perfect so far.

Anyway, thanks for the insight, I will fix all your comments.